libs/blocks/merch/merch.js

```diff
@@ -40,70 +27,88 @@ function omitNullValues(target) {
   return target;
 }
 
-const getTacocatEnv = (envName, locale) => {
-  const wcsLocale = GEO_MAPPINGS[locale.prefix] ?? locale.ietf;
-  // eslint-disable-next-line prefer-const
-  let [language, country = 'US'] = wcsLocale.split('-', 2);
+const imsCountryPromise = new Promise((resolve) => {
+  let count = 0;
+  const check = setInterval(() => {
+    count += 1;
+    if (window.adobeIMS) {
+      clearInterval(check);
+      if (window.adobeIMS.isSignedInUser()) {
+        window.adobeIMS
+          .getProfile()
+          .then(({ countryCode }) => resolve(countryCode))
+          .catch(() => resolve());
+      } else {
+        resolve();
+      }
+    } else if (count > 25) {
+      clearInterval(check);
+      resolve();
+    }
+  }, 200);
+});
+
+export const getTacocatEnv = (envName, locale) => {
+  const wcsLocale = (GEO_MAPPINGS[locale.prefix] ?? locale.ietf).split('-', 2);
+  let language = wcsLocale[0];
+  const country = wcsLocale[1] || 'US';
   if (!SUPPORTED_LANGS.includes(language)) {
     language = 'en';
   }
-  const host = envName === ENV_PROD
-    ? 'https://www.adobe.com'
-    : 'https://www.stage.adobe.com';
-  const scriptUrl = `${host}/special/tacocat/lib/${VERSION}/tacocat.js`;
+  const host = envName === ENV_LOCAL
+    ? 'https://www.stage.adobe.com'
+    : 'https://www.adobe.com';
+
   const literalScriptUrl = `${host}/special/tacocat/literals/${language}.js`;
-  return { scriptUrl, literalScriptUrl, country, language };
+  const scriptUrl = `${host}/special/tacocat/lib/${VERSION}/tacocat.js`;
+  const tacocatEnv = envName === ENV_LOCAL ? 'STAGE' : 'PRODUCTION';
```

Comment:
Same as above, a non `local` env is automatically treated as `prod`.

Reply:
as explained above, we can discuss this in the milo commerce meeting.